bam_sort.c

```diff
@@ -911,6 +912,36 @@ int* rtrans_build(int n, int n_targets, trans_tbl_t* translation_tbl)
 #define MERGE_COMBINE_PG 32 // Combine PG tags frather than redefining them
 #define MERGE_FIRST_CO   64 // Use only first file's @CO headers (sort cmd only)
 
+
+static hts_reglist_t *duplicate_reglist(const hts_reglist_t *rl, int rn) {
+    if (!rl)
+        return NULL;
+
+    hts_reglist_t *new_rl = calloc(rn, sizeof(hts_reglist_t));
+    if (!new_rl)
+        return NULL;
+
+    int i;
+    for (i=0; i < rn; i++) {
+        new_rl[i].tid     = rl[i].tid;
+        new_rl[i].count   = rl[i].count;
+        new_rl[i].min_beg = rl[i].min_beg;
+        new_rl[i].max_end = rl[i].max_end;
+
+        new_rl[i].reg = rl[i].reg;
+        new_rl[i].intervals = calloc(new_rl[i].count, sizeof(hts_pair_pos_t));
+        if (!new_rl[i].intervals) {
+            hts_reglist_free(new_rl, i);
+            return NULL;
+        }
+        int j;
```

Comment:
Would be quicker and possibly simpler to do `memcpy(new_rl[i].intervals, rl[i].intervals, sizeof(new_rl[i].intervals[0]) * count)` here.